docs/_packages/oauth.md

```diff
@@ -62,17 +62,47 @@ const installer = new InstallProvider({
 
 ---
 
-### Generating an installation URL
+## Showing an Installation Page
 
-You'll need an installation URL when you want to test your own installation, in order to submit your app to the App Directory, and if you need an additional authorizations (user tokens) from users inside a team when your app is already installed. These URLs are also commonly used on your own webpages as the link for an ["Add to Slack" button](https://api.slack.com/docs/slack-button). You may also need to generate an installation URL dynamically when an option's value is only known at runtime, and in this case you would redirect the user to the installation URL.
+You'll need an installation URL when you want to test your own installation in order to submit your app to the App Directory and in case you need additional authorizations (such as user tokens) from users inside a team where your app is already installed. These URLs are also commonly used on your own webpages as the link for an ["Add to Slack" button](https://api.slack.com/docs/slack-button).
 
-The `installProvider.generateInstallUrl()` method will create an installation URL for you. It takes in an options argument which at a minimum contains a `scopes` property. `installProvider.generateInstallUrl()` options argument also supports `metadata`, `teamId`, `redirectUri` and `userScopes` properties.
+There are two approaches the `InstallProvider` provides to show an installation page to the user prior to beginning the OAuth process:
+
+1. You can let `InstallProvider` render the installation page at a URL/path of your choosing [using the `handleInstallPath()` method](#using-handleinstallpath). It will automatically display an "Add to Slack" button and encode any desired user or bot scopes and metadata you specify.
+2. You can render the installation page yourself and [call `generateInstallUrl()`](#using-generateinstallurl) to generate a secure slack.com-based URL to initiate the OAuth process. You would use this generated URL in your own installation page's links/buttons/UI elements.
```

Comment:
You can customize the behavior by passing `renderHtmlForInstallPath` function when initializing `InstallProvider`. Thus, there is no limitation on it in `handleInstallPath`. We can say that "For most use cases, we recommend going with `handleInstallPath()` with necessary customize options. If you need only the authorize URL generation, you can use `generateInstallUrl()`, which is called inside `handleInstallPath()`, in your code.

```suggestion
1. You can let `InstallProvider` render the installation page at a URL/path of your choosing [using the `handleInstallPath()` method](#using-handleinstallpath). It will automatically display an "Add to Slack" button and encode any desired user or bot scopes and metadata you specify.
2. You can customize the installation page by passing `renderHtmlForInstallPath` function to the `InstallProvider` constructor. Also, if your app supports Direct Install URL in the App Directory page, you can pass `directInstall: true` when initializing `InstallProvider`.
```